File: app/src/main/java/com/alvarlagerlof/koda/WebClient.java
package com.alvarlagerlof.koda;

import android.content.Context;
import android.content.DialogInterface;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.webkit.JsPromptResult;
import android.webkit.JsResult;
import android.webkit.WebView;
import android.widget.EditText;
import android.widget.TextView;

/**
 * Created by alvar on 2016-11-08.
 */

public class WebClient {

    public android.webkit.WebChromeClient getClient(final Context context, final String title) {

        final String alertTitle = title + " " + context.getString(R.string.says) + "...";

        final android.webkit.WebChromeClient webChromeClient = new android.webkit.WebChromeClient(){

            @Override
            public boolean onJsAlert(WebView view, String url, String message, final JsResult result) {
                new AlertDialog.Builder(context)
                        .setTitle(alertTitle)
                        .setMessage(message)
                        .setPositiveButton(android.R.string.ok,
                                new AlertDialog.OnClickListener() {
                                    public void onClick(DialogInterface dialog, int which) {
                                        result.confirm();
                                    }
                                }).setCancelable(false).create().show();

                return true;
            }

            @Override
            public boolean onJsConfirm(WebView view, String url, String message, final JsResult result) {
                new AlertDialog.Builder(context)
                        .setTitle(alertTitle)
                        .setMessage(message)
                        .setPositiveButton(android.R.string.ok,
                                new DialogInterface.OnClickListener() {
                                    public void onClick(DialogInterface dialog, int which) {
                                        result.confirm();
                                    }
                                }).setNegativeButton(android.R.string.cancel,
                        new DialogInterface.OnClickListener() {
                            public void onClick(DialogInterface dialog, int which) {
                                result.cancel();
                            }
                        }).create().show();
                return true;
            }

            @Override
            public boolean onJsPrompt(WebView view, String url, String message, String defaultValue, final JsPromptResult result) {
                final LayoutInflater factory = LayoutInflater.from(context);
                final View v = factory.inflate(R.layout.javascript_prompt_dialog, null);

                ((TextView)v.findViewById(R.id.prompt_message_text)).setText(message);

                new AlertDialog.Builder(context)
                        .setTitle(alertTitle)
                        .setView(v)
                        .setPositiveButton(android.R.string.ok,
                                new DialogInterface.OnClickListener() {
                                    public void onClick(DialogInterface dialog, int whichButton) {
                                        String value = ((EditText)v.findViewById(R.id.prompt_input_field)).getText().toString();
                                        result.confirm(value);
                                    }
                                })
                        .setNegativeButton(android.R.string.cancel,
                                new DialogInterface.OnClickListener() {
                                    public void onClick(DialogInterface dialog, int whichButton) {
                                        result.cancel();
                                    }
                                })
                        .setOnCancelListener(
                                new DialogInterface.OnCancelListener() {
                                    public void onCancel(DialogInterface dialog) {
                                        result.cancel();
                                    }
                                })
                        .show();

                return true;
            }

        };

        return webChromeClient;
    }

}

File: app/src/main/java/com/alvarlagerlof/koda/WebClient.kt
package com.alvarlagerlof.koda

import android.content.Context
import android.support.v7.app.AlertDialog
import android.view.LayoutInflater
import android.webkit.JsPromptResult
import android.webkit.JsResult
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.widget.EditText
import android.widget.TextView
import com.alvarlagerlof.koda.Extensions.hideKeyboard

/**
 * Created by alvar on 2017-06-18.
 */

class WebClient(internal var context: Context, internal var title: String) : WebChromeClient() {

    val alertTitle = title

    override fun onJsAlert(view: WebView, url: String, message: String, result: JsResult): Boolean {
        AlertDialog.Builder(context)
                .setTitle(alertTitle)
                .setMessage(message)
                .setPositiveButton(android.R.string.ok) { _, _ ->
                    view.hideKeyboard()
                    result.confirm()
                }
                .setCancelable(false)
                .create()
                .show()

        return true
    }

    override fun onJsConfirm(view: WebView, url: String, message: String, result: JsResult): Boolean {
        AlertDialog.Builder(context)
                .setTitle(alertTitle)
                .setMessage(message)
                .setPositiveButton(android.R.string.ok) { _, _ ->
                    view.hideKeyboard()
                    result.confirm()
                }
                .setNegativeButton(android.R.string.cancel) { _, _ ->
                    view.hideKeyboard()
                    result.cancel()
                }
                .setCancelable(false)
                .create()
                .show()
        return true
    }

    override fun onJsPrompt(view: WebView, url: String, message: String, defaultValue: String, result: JsPromptResult): Boolean {
        val factory = LayoutInflater.from(context)
        val v = factory.inflate(R.layout.javascript_prompt_dialog, null)

        val prompt_message_text: TextView = v.findViewById(R.id.prompt_message_text)
        val prompt_input_field: EditText = v.findViewById(R.id.prompt_input_field)
        prompt_message_text.text = message

        AlertDialog.Builder(context)
                .setTitle(alertTitle)
                .setView(v)
                .setPositiveButton(android.R.string.ok) { _, _ ->
                    val value = prompt_input_field.text.toString()
                    view.hideKeyboard()
                    result.confirm(value)
                }
                .setNegativeButton(android.R.string.cancel) { _, _ ->
                    view.hideKeyboard()
                    result.cancel()
                }
                .setCancelable(false)
                .show()

        return true
    }
}
